src/mpyl/steps/deploy/k8s/kubectl.py

```diff
@@ -0,0 +1,85 @@
+""" This module is called on to push your kubernetes manifest to a(n) (argo) repo during the `mpyl.steps.deploy step."""
+
+import shutil
+from logging import Logger
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from .rancher import ClusterConfig
+from ....project import Target
+from ....utilities.repo import RepoConfig, Repository
+from ....steps import Input, Output
+from ....utilities.subprocess import custom_check_output
+
+
+def push_manifest_to_repo(
+    logger: Logger,
+    step_input: Input,
+    rancher_config: ClusterConfig,
+    manifest_path: Path,
+) -> Output:
+    git_config = step_input.run_properties.config["vcs"]["argoRepository"]
+    if not git_config:
+        raise ValueError("No argocd repository configured")
+    argocd_repo_config = RepoConfig.from_git_config(git_config=git_config)
+    with TemporaryDirectory() as tmp_repo_dir:
+        with Repository.from_clone(
+            config=argocd_repo_config, repo_path=Path(tmp_repo_dir)
+        ) as artifact_repo:
+            branch = "feature/TECH-610-implement-argocd-2"  # This can be main later
+            if artifact_repo.local_branch_exists(branch_name=branch):
+                artifact_repo.delete_local_branch(
+                    branch_name=branch
+                )  # To enforce latest version of branch
+            branch_exists = artifact_repo.remote_branch_exists(branch_name=branch)
+            if branch_exists:
+                artifact_repo.checkout_branch(branch_name=branch)
+            else:
+                artifact_repo.create_branch(branch_name=branch)
+
+            folder_name = __get_folder_name(step_input=step_input)
+            namespace = (
+                step_input.project.deployment.namespace
+                if step_input.project.deployment
+                else None
+            )
+            new_file_path = Path(
+                tmp_repo_dir,
+                namespace or "",
+                step_input.project.name,
+                folder_name,
+                manifest_path.name,
+            )
+            shutil.copytree(
+                src=manifest_path.parent,
+                dst=new_file_path.parent,
+                dirs_exist_ok=True,
+            )
+
+            #  validate manifest
+            validate_command = (
```

Comment:
Maybe good to think a bit on this approach. Once we're fully on argo, I think it would make sense to get rid of as much of the k8s related concepts from mpyl as possible. No helm, no rancher config etc.

Reply:
I don't see any harm in using kubectl? It can be used for so much more than just deploying. Besides, how else do we make sure we push valid manifests to the *main* branch of the argo repo?

Reply:
I agree with @koeves 👍 

Instead of pushing directly to the `main` branch, we can create a PR, configure an action to automatically validate, lint, and whatever else we need, and automatically merge to main when green. I used this exact same approach in the past and it worked _really well_.

This also solves the problem of race conditions with multiple deployments: using the current approach of pushing to `main` will work, but we'll immediately start seeing a lot of 
```
error: failed to push some refs to '<repository>'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

once every project starts deploying at same time.

Reply:
I forced a schema error https://github.com/SamTheisens/mpyl-example-argocd/pull/3/commits/53d5e9bfecfb0277531ad423465652c2c2a20d81

Not the clearest error, but the deployment fails:
![Screenshot 2023-11-03 at 20 23 15](https://github.com/Vandebron/mpyl/assets/1911436/8bb84796-fbda-4e35-a32e-f2380f1a42d4)